packages/storage-plus/src/indexes.rs

```diff
@@ -106,6 +121,10 @@ where
         Box::new(mapped)
     }
 
+    pub fn prefix(&self, p: K::Prefix) -> Prefix<T> {
```

Comment:
Okay, this may be the issue - we need to see the remainder of the key used in the Prefix. Does this work in primary keys?

Reply:
This only works when you pass the complete index_function result to `prefix`.

See commented [line 399](https://github.com/CosmWasm/cosmwasm-plus/blob/7f8c315bd38f9726e4d19181844bb3a802dfde47/packages/storage-plus/src/indexed_map.rs#L399)